diff -ur apache-ant-1.6.5/src/main/org/apache/tools/ant/taskdefs/Exec.java apache-ant-1.6.5-antRun/src/main/org/apache/tools/ant/taskdefs/Exec.java
--- apache-ant-1.6.5/src/main/org/apache/tools/ant/taskdefs/Exec.java	2005-06-02 15:19:56.000000000 +0200
+++ apache-ant-1.6.5-antRun/src/main/org/apache/tools/ant/taskdefs/Exec.java	2006-09-12 14:57:55.000000000 +0200
@@ -99,7 +99,7 @@
                             + "found", getLocation());
                     }
 
-                    String antRun = getProject().resolveFile(ant + "/bin/antRun.bat").toString();
+                    String antRun = getProject().resolveFile("@BINDIR@/antRun.bat").toString();
                     command = antRun + " " + dir + " " + command;
                 }
             }

