--- jakarta-ant-1.4.1/src/script/ant.wiget	Thu Oct 11 15:58:30 2001
+++ jakarta-ant-1.4.1/src/script/ant	Thu Mar 21 15:02:29 2002
@@ -14,13 +14,13 @@
 
 if [ -z "$ANT_HOME" ] ; then
   # try to find ANT
-  if [ -d /opt/ant ] ; then 
+  if [ -f /usr/share/java/ant.jar ] ; then 
+    ANT_HOME=/usr/share/java
+  elif [ -d /opt/ant ] ; then 
     ANT_HOME=/opt/ant
-  fi
-
-  if [ -d ${HOME}/opt/ant ] ; then 
+  elif [ -d ${HOME}/opt/ant ] ; then 
     ANT_HOME=${HOME}/opt/ant
-  fi
+  else	
 
   ## resolve links - $0 may be a link to ant's home
   PRG=$0
@@ -37,7 +37,7 @@
   done
   
   ANT_HOME=`dirname "$PRG"`/..
-
+  fi
 fi
 
 # For Cygwin, ensure paths are in UNIX format before anything is touched
