--- apache-ant-1.6.3/src/script/ant.orig	2005-04-28 15:43:12.000000000 +0200
+++ apache-ant-1.6.3/src/script/ant	2005-05-07 11:48:33.504821320 +0200
@@ -87,6 +87,9 @@
 esac
 
 if [ -z "$ANT_HOME" -o ! -d "$ANT_HOME" ] ; then
+  if [ -f /usr/share/java/ant.jar ]; then
+    ANT_HOME=/usr/share/java
+  else
   ## resolve links - $0 may be a link to ant's home
   PRG="$0"
   progname=`basename "$0"`
@@ -106,6 +109,7 @@
 
   # make it fully qualified
   ANT_HOME=`cd "$ANT_HOME" && pwd`
+  fi
 fi
 
 # For Cygwin, ensure paths are in UNIX format before anything is touched
